github-actions/feature-request/src/action.ts

```diff
@@ -0,0 +1,218 @@
+ * posted from the action. This way we track when the voting
+ * process has started, when we have posted a warning message,
+ * and when it's time to move to the next stage.
+ */
+export enum CommentMarkers {
+  StartVoting = '<!-- 6374bc4f-3ca6-4ebb-b416-250033c91ab5 -->',
+  Warn = '<!-- 727acbae-59f4-4cde-b59e-4c9847cabcca -->',
+  Close = '<!-- 5bb7b168-3bff-4685-ac2a-be3174a97af4 -->',
+}
+
+export interface Config {
+  organization: string;
+
+  featureRequestLabel: string;
+  requiresVotesLabel: string;
+  inBacklogLabel: string;
+  underConsiderationLabel: string;
+
+  // For old issues we don't want to go to the standard
+  // process because they have been already open for long enough.
+  // Instead we want to wait `oldIssuesDaysWarnDuration` number of days
+  // until we post a warning and close it in `closeAfterWarnDaysDuration`.
+  oldIssueWarnDaysDuration: number;
+  // This flag indicates for how long after the initial comment
+  // that kicks off the voting process we should post a warning comment.
+  warnDaysDuration: number;
+  // Indicates how may days after the warning comment we should close
+  // the issue and post an explanation.
+  closeAfterWarnDaysDuration: number;
+
+  closeComment: string;
+  warnComment: string;
+  startVotingComment: string;
+
+  // Number of votes up required to add the `underConsiderationLabel`.
+  minimumVotesForConsideration: number;
+  // Number of comments from unique authors required to add the `underConsiderationLabel`.
+  // Keep in mind that this includes comments from the bot and team members.
+  minimumUniqueCommentAuthorsForConsideration: number;
+
+  // If set to true, the bot will automatically close the issue
+  // when voting has ended, if the request has not collected a sufficient number of votes.
+  // Alternatively, the bot will just add a `votingFinishedLabel`.
+  closeWhenNoSufficientVotes: boolean;
+  insufficientVotesLabel: string;
+}
+
+export const run = async (api: GitHubAPI, config: Config) => {
+  const issues = api.query({
+    q: `is:open is:issue label:"${config.featureRequestLabel}" -label:"${config.inBacklogLabel}" -label:"${config.underConsiderationLabel}" sort:created-asc`,
+  });
+
+  for await (const issue of issues) {
+    await processIssue(api, issue, config);
+  }
+};
+
+/**
+ * Processes a particular issue from the repository. At the beginning of the invocation
+ * the function makes sure the issue is in the target group of feature requests we want to process.
+ * Depending on the current state of the issue, the function will either label it, close it,
+ * add a comment, or perform a noop.
+ *
+ * @param githubAPI Object used for querying the GitHub API
+ * @param githubIssue Object used for communication with the GitHub API for a particular issue
+ * @param config Configuration of the GitHub bot
+ */
+const processIssue = async (githubAPI: GitHubAPI, githubIssue: GitHubIssueAPI, config: Config) => {
```

Comment:
Is there a reason to have this as a `const` rather than a declared local function?

Reply:
A practice I use to reduce the risk of overriding a function if I have a second declaration with the same name.